edge-hub/src/Microsoft.Azure.Devices.Routing.Core/Route.cs

```diff
@@ -45,7 +51,7 @@ public bool Equals(Route other)
             return string.Equals(this.Id, other.Id) &&
                    string.Equals(this.Condition, other.Condition) &&
                    this.Source.Equals(other.Source) &&
-                   this.Endpoints.SetEquals(other.Endpoints);
+                   this.Endpoint.Equals(other.Endpoint);
```

Comment:
1. Do you know why IoTHubName is not considered in Equals method?
2. Also IMessageSource doesn't implement IEquatable or similar interface, are we doing reference comparison here?
3. For string comparison, should we pass string comparison option?

Reply:
I'm not clear on the history here, but if we want to make improvements to the Endpoint implementation we can do it in a separate PR.

---
In reply to: [377446071](https://github.com/Azure/iotedge/pull/2469#discussion_r377446071) [](ancestors = 377446071)